mvpa2/measures/label.py

```diff
@@ -0,0 +1,107 @@
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the PyMVPA package for the
+#   copyright and license terms.
+#
+### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Label adjacent features in a dataset
+"""
+import numpy as np
+
+from ..base.param import Parameter
+from ..base.constraints import EnsureBool
+from ..datasets.base import Dataset
+from ..misc.neighborhood import CachedQueryEngine
+
+from .base import Measure
+
+if __debug__:
+    from ..base import debug
+
+
+class Labeler(Measure):
+    """Sensitivities of features for a given Classifier.
+
+    """
+    qe = Parameter(
+        None,
+        doc="QueryEngine to determine neighbors of any given feature")
+
+    cache = Parameter(
+        True,
+        constraints=EnsureBool(),
+        doc="Wrap provided QueryEngine with CachedQueryEngine to speed up "
+            "subsequent queries"
+    )
+
+    def __init__(self, qe, space='maxlabels', **kwargs):
+        """Initialize labeler with a query engine
+
+        Parameters
+        ----------
+        qe : `QueryEngine`
+          Query engine to use.
+        """
+        if kwargs.get('cache', True) and not isinstance(qe, CachedQueryEngine):
+            qe = CachedQueryEngine(qe)
+        super(Labeler, self).__init__(qe=qe, space=space, **kwargs)
+        self._untrain()
+
+    @property
+    def is_trained(self):
+        return self._nfeatures_trained is not None
+
+    def _train(self, ds):
+        self.params.qe.train(ds)
+        self._nfeatures_trained = ds.nfeatures
```

Comment:
Maybe it's safer to store the entire .fa? Then one could detect cases with the same number of features but still geometry.